protos/core.proto

```diff
@@ -37,7 +37,12 @@ message Environments {
       }
     }
   
-    GitReference git_reference = 2;
+    oneof git_settings {
+      string git = 5;
```

Comment:
this doesn't feel right to me, it should really probably be `git_reference`, `reference`, `ref`, `git_ref` or something? Can we rename some stuff?

Reply:
Amusingly I actually started with `git_ref` but changed my mind. Going with that...